theatre/core/src/propTypes/internals.ts

```diff
@@ -47,12 +48,31 @@ export type ShorthandPropToLonghandProp<P extends IShorthandProp> =
     ? PropTypeConfig_Boolean
     : P extends PropTypeConfig
     ? P
-    : P extends IShorthandCompoundProps
+    : P extends UnknownShorthandCompoundProps
     ? PropTypeConfig_Compound<ShorthandCompoundPropsToLonghandCompoundProps<P>>
     : never
 
+export type ShorthandCompoundPropsToInitialValue<
+  P extends UnknownShorthandCompoundProps,
+> = LonghandCompoundPropsToInitialValue<
+  ShorthandCompoundPropsToLonghandCompoundProps<P>
+>
+
+type LonghandCompoundPropsToInitialValue<P extends UnknownValidCompoundProps> =
+  {
+    [K in keyof P]: P[K]['valueType']
+  }
+
+export type PropsValue<P> = P extends UnknownValidCompoundProps
+  ? LonghandCompoundPropsToInitialValue<P>
+  : P extends UnknownShorthandCompoundProps
+  ? LonghandCompoundPropsToInitialValue<
+      ShorthandCompoundPropsToLonghandCompoundProps<P>
+    >
+  : never
```

Comment:
These can just use `ShorthandCompoundPropsToLonghandCompoundProps<P>['valueType']` and be removed

Reply:
Note: `PropsValue<P>` is a bit more flexible than `ShorthandCompoundPropsToLonghandCompoundProps<P>`, because it accepts a dict to prop configs OR a dict to shorthand props.